### PR TITLE
Enhancements/component sweep

### LINKED ISSUE #296: Convert components to React.createClass
We prefer React.createClass so let's get rid of all the horrible ES6 extends :)

### PR DESCRIPTION
**Pull request by ch2ch3**
@ustwo/usweb-dev @collingo 

- use `React.createClass()` everywhere
- use double quotes on HTML attributes

(Fixes #296)

### CODE REVIEW

**Comment by daaain** [comment]
LGTM on a quick scroll around in the changes, anything specific to check?

**Comment by ch2ch3** [comment]
If you don't mind, have a look at the components that are affected by the new mixins :)

File: src/app/components/app/index.js [hunk 1]
```diff
@@ -37,104 +38,41 @@ const pageMap = {
...
+    return React.createElement(pageMap[pageId], this.state);
+  },
   renderModal() {
-    const state = this.state;
-    const takeover = state.takeover;
-    const modalType = state.modal;
+    const { takeover, modal: modalType } = this.state;
     let modal;
-    if(this.showTakeover()) {
+    if (this.showTakeover()) {
```
Comment (davidey) [inline src/app/components/app/index.js]: What's the convention for this?
I'm for the `if (...)` over the `if(...)`, but I see the latter quite often.
Let's agree on a standard and use it @ustwo/usweb-dev @collingo

Comment (daaain) [inline src/app/components/app/index.js]: We have 29 `if(`s and 107 `if (`s so the latter is the winner, but it's a good point, maybe this branch is a good chance to get that consistent.

**Comment by davidey** [comment]
Massive branch! Wonder if it was splittable into different sub-branches? Eg. first take care of the `createClass`, then the double quotes, then the other general cleanup. Would make it easy to review them. 
Let's do the same with the `if (...)`, so it's easier to spot them clearly in the PR review.

Anyway, looks good to me, if it's been tested throughout :)

File: src/app/components/app/index.js (hunk 1 above)
Comment (ch2ch3) [inline src/app/components/app/index.js]: @davidey my personal preference is for `if (...)` too, anyway found this from our CONTRIBUTING.md https://github.com/airbnb/javascript#18.3

**Comment by daaain** [comment]
Oh yeah, I did have a look at `SVGSequenceAnimation` and this mixin way is even nicer, much more obvious what's going on.

File: src/app/components/app/index.js (hunk 1 above)
Comment (arnau) [inline src/app/components/app/index.js]: I have strong feelings against `if(`

Comment (daaain) [inline src/app/components/app/index.js]: Time to introduce a linter on the CI? :trollface:

Comment (ch2ch3) [inline src/app/components/app/index.js]: https://houndci.com/? :smile:

**Pull request merged**